src-plugins/itkFilters/itkFiltersGaussianProcess.cpp

```diff
@@ -14,36 +14,36 @@
 #include <itkFiltersGaussianProcess.h>
 
 #include <dtkCore/dtkAbstractProcessFactory.h>
-#include <medAbstractDataFactory.h>
-
-#include <medMetaDataKeys.h>
 
 #include <itkImage.h>
-#include <itkCommand.h>
 #include <itkSmoothingRecursiveGaussianImageFilter.h>
 
-#include <itkExceptionObject.h>
-#include <itkFiltersGaussianProcess_p.h>
+#include <medUtilities.h>
+
+class itkFiltersGaussianProcessPrivate
+{
+public:
+    double sigma;
+};
+
+const double itkFiltersGaussianProcess::defaultSigma = 1.0;
 
 itkFiltersGaussianProcess::itkFiltersGaussianProcess(itkFiltersGaussianProcess *parent) 
-    : itkFiltersProcessBase(*new itkFiltersGaussianProcessPrivate(this), parent)
+    : itkFiltersProcessBase(parent), d(new itkFiltersGaussianProcessPrivate)
 {
-    DTK_D(itkFiltersGaussianProcess);
-    
-    d->filter = this;
-    d->output = NULL; 
-    
-     d->description = tr("ITK gaussian filter");
+    d->sigma = defaultSigma;
 }
 
-
-itkFiltersGaussianProcess::itkFiltersGaussianProcess(const itkFiltersGaussianProcess& other) 
-    : itkFiltersProcessBase(*new itkFiltersGaussianProcessPrivate(*other.d_func()), other)
+itkFiltersGaussianProcess::itkFiltersGaussianProcess(const itkFiltersGaussianProcess& other)
+     : itkFiltersProcessBase(other), d(other.d)
```

Comment:
Could have been removed, no ?

Reply:
I asked @mathildemerle to keep the copy constructors because they were already there and it's always good to have proper copy constructors (especially since dtkAbstractObject provides an explicit copy constructor AND a copy function, so all derived classes not implementing copy constructors are faulty (which includes 99% of the classes)).

Reply:
(the default one is not sufficient because we have private implementation pointers in our classes)